graylog2-web-interface/src/views/pages/EventDefinitionReplaySearchPage.test.tsx

```diff
@@ -0,0 +1,117 @@
+import EventDefinitionReplaySearchPage from 'views/pages/EventDefinitionReplaySearchPage';
+import useEventDefinition from 'hooks/useEventDefinition';
+import useAlertAndEventDefinitionData from 'hooks/useAlertAndEventDefinitionData';
+import {
+  mockedMappedAggregation,
+  mockEventDefinitionTwoAggregations,
+} from 'helpers/mocking/EventAndEventDefinitions_mock';
+import useParams from 'routing/useParams';
+
+const mockView = createSearch();
+
+jest.mock('views/components/Search');
+jest.mock('routing/useParams');
+
+jest.mock('views/logic/views/Actions');
+jest.mock('views/logic/views/UseCreateViewForEventDefinition');
+jest.mock('views/logic/views/UseProcessHooksForView');
+jest.mock('views/hooks/useLoadView');
+
+jest.mock('hooks/useEventDefinition');
+jest.mock('hooks/useAlertAndEventDefinitionData');
+
+jest.mock('stores/event-notifications/EventNotificationsStore', () => ({
+  EventNotificationsActions: {
+    listAll: jest.fn(async () => Promise.resolve),
+  },
+  EventNotificationsStore: MockStore((['getInitialState', () => ({ all: [] })])),
+}));
+
+jest.mock('views/logic/Widgets', () => ({
+  ...jest.requireActual('views/logic/Widgets'),
+  widgetDefinition: () => ({
+    searchTypes: () => [{
+      type: 'AGGREGATION',
+      typeDefinition: {},
+    }],
+  }),
+}));
+
+describe('EventDefinitionReplaySearchPage', () => {
+  const SimpleReplaySearchPage = () => (
+    <StreamsContext.Provider value={[{}]}>
+      <EventDefinitionReplaySearchPage />
+    </StreamsContext.Provider>
+  );
+
+  beforeAll(loadViewsPlugin);
+
+  afterAll(unloadViewsPlugin);
+
+  beforeEach(() => {
+    asMock(useParams).mockReturnValue({ definitionId: mockEventDefinitionTwoAggregations.id });
+    asMock(UseCreateViewForEventDefinition).mockReturnValue(Promise.resolve(mockView));
+    asMock(useProcessHooksForView).mockReturnValue({ status: 'loaded', view: mockView, executionState: SearchExecutionState.empty() });
+    asMock(SearchComponent).mockImplementation(() => <span>Extended Search Page</span>);
+
+    asMock(useEventDefinition).mockImplementation(() => ({
+      data: { eventDefinition: mockEventDefinitionTwoAggregations, aggregations: mockedMappedAggregation },
+      isLoading: false,
+      isFetched: true,
+      refetch: () => {},
+    }));
+  });
+
+  it('should run useEventDefinition, UseCreateViewForEvent with correct parameters', async () => {
+    asMock(useAlertAndEventDefinitionData).mockImplementation(() => ({
+      eventData: undefined,
+      eventDefinition: mockEventDefinitionTwoAggregations,
+      aggregations: mockedMappedAggregation,
+      isEvent: false,
+      isEventDefinition: true,
+      isAlert: false,
+      alertId: undefined,
+      definitionId: mockEventDefinitionTwoAggregations.id,
+      definitionTitle: mockEventDefinitionTwoAggregations.title,
+    }));
+
+    // eslint-disable-next-line testing-library/no-unnecessary-act
+    await act(async () => {
+      render(<SimpleReplaySearchPage />);
+    });
```

Comment:
Is it may possible to not use act here, but wait for `useEventDefinition` to have been called?